### PR TITLE
git(hooks): Add warning to prevent commit with wrong syntaxes

### PR DESCRIPTION
**Pull request by oumar-fall**
### Description of change

The commits adds a hook to prevent developers from commiting changes in a wrong format which would prevent a new version of the package to be released

### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] `yarn lint` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

**Review by julien-devatom — APPROVED** [review]

**Review by julien-devatom — APPROVED** [review]
you are missing the optional "!" for breaking changes, like
```bash
feat(subject)!: breaking change
```

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+COMMIT_MSG_FILE="$1"
+
+exec < /dev/tty
+
+if ! grep -iqE "((feat)|(fix)|(build))\([a-zA-Z0-9]+\):.+" "$COMMIT_MSG_FILE"; then
```
Comment (Rubilmax) [inline .husky/commit-msg:7]: `build` triggers a release?

**Review by Rubilmax — COMMENTED** [review]
I believe commitlint is better suited for this, though you won't have a warning that the commit won't trigger a release. As it's not my dev env, i don't care

File: .husky/commit-msg (hunk 1 above)
Comment (oumar-fall) [reply .husky/commit-msg:7]: obviously 
image

File: .husky/commit-msg [hunk 2]
```diff
@@ -4,7 +4,7 @@ COMMIT_MSG_FILE="$1"
 
 exec < /dev/tty
 
-if ! grep -iqE "((feat)|(fix)|(build))\([a-zA-Z0-9]+\):.+" "$COMMIT_MSG_FILE"; then
+if ! grep -iqE "((feat)|(fix)|(build))\([a-zA-Z0-9_-]+\)\!?:.+" "$COMMIT_MSG_FILE"; then
```
Comment (Rubilmax) [inline .husky/commit-msg:7]:
```suggestion
if ! grep -iqE "(feat|fix|build)(\([a-zA-Z0-9_-]+\))?\!?:.+" "$COMMIT_MSG_FILE"; then
```

the parenthesis part is optional in conventional commits

**Review by Rubilmax — APPROVED** [review]

**Pull request merged**